geo/src/main/java/org/odk/collect/geo/DaggerSetup.kt

```diff
@@ -112,4 +108,14 @@ open class GeoDependencyModule {
             }
         }
     }
+
+    @Provides
+    open fun providesOfflineMapLayersPickerViewModelFactory(): OfflineMapLayersPickerViewModel.Factory {
```

Comment:
I think I'd prefer it if we just added providers for `ReferenceLayerRepository` and `SettingsProvider` here. We could then avoid a specific implementation of `ViewModelProvider.Factory` for `OfflineMapLayersPickerViewModel` and just have a shared view model factory for geo components that can construct one (like we do with form entry). This approach keeps a more specific set of components "injectable" (the "building blocks") and should reduce the amount of view model factories we need.

Reply:
> I think I'd prefer it if we just added providers for ReferenceLayerRepository and SettingsProvider here. We could then avoid a specific implementation of ViewModelProvider.Factory for OfflineMapLayersPickerViewModel and just have a shared view model factory for geo components that can construct one (like we do with form entry). 

I see what you mean. I think the reason I did this in that way was that I wanted to have one class to wrap the three dependencies I need (`ReferenceLayerRepository`, `Scheduler`, `SettingsProvider` and potentially more when I add importing/deleting layers) so that I don't have to inject all of them separately in `GeoPolyActivity`, `GeoPointMapActivity`, `SelectionMapFragment` and `MapsPreferencesFragment` (every place where we use the new maps picker). Do you think converting `OfflineMapLayersPickerViewModel.Factory` to a more general one (but still injecting it in the same way) that would in the future as you said work like `FormEntryViewModelFactory` would be a good trade off here?

Reply:
> I think the reason I did this in that way was that I wanted to have one class to wrap the three dependencies I need (ReferenceLayerRepository, Scheduler, SettingsProvider and potentially more when I add importing/deleting layers) so that I don't have to inject all of them separately in GeoPolyActivity, GeoPointMapActivity, SelectionMapFragment and MapsPreferencesFragment (every place where we use the new maps picker).

I think it's ok to have all those injections. I like the model of dependency injection just handling the edges of the system. The hope there is you end up with more being injected into entry point "black boxes" (like Activity objects), but less actual depdencies being managed by the framework with plain ol' constructors doing most of the work.

Maybe a nice way of having our cake and eating it here is to create two `Factory` implementations - one for `geo` (`GeoViewModelFactory`) and one for `MapPreferenecesFragment` in `collect_app` (`MapPreferencesViewModelFactory`). Then we could just use `@Inject` on their constructors. You still only have to deal with one `@Inject` in both places, but we still don't have declare the factory as a dependency of `geo` anywhere. I'd thought about doing this for `FormEntryViewModelFactory`  before, but hadn't got round to it. I like this approach as it gives us the convenience of Dagger in the Activity/Fragment, but doesn't mean managing dependencies outside (in the module etc) and you still have a simple ViewModel factory constructor.

Reply:
I've decided to go for the first option.